src/bio.h

```diff
@@ -10,6 +10,8 @@
 #ifndef __BIO_H
 #define __BIO_H
 
+#include <stdint.h>
```

Comment:
what is it for?

Reply:
It's for `uint64_t`.

Reply:
> Did you have a compilation error?

No.

> Theoretically, it was introduced in server.h.

Quite a few `.c` files include `bio.h` 
```
aof.c
bio.c
config.c
db.c
debug.c
evict.c
lazyfree.c
rdb.c
replication.c
server.c
```
and `bio.h` doesn't include anything. Shouldn't it be self-contained in accordance with "include what you use"? Why rely on that every user of the header has included `stdint.h` one way or the other? Why rely on inclusion _order_ to make it work? Every user of `bio.h` must now do it in this order:
```
#include "server.h"
#include "bio.h"
```
Why force the inclusion of `server.h`? Theoretically, not all those `.c` files need it.

Reply:
In Redis, `bio.h` is not intended to be a standalone header. It is always used
together with `server.c` as part of the server module. Any `.c` file that needs
`bio.h` will also include `server.h`.
It is not like the relatively independent modules of listpack quickilst dict. For them, self-contained is necessary.

Reply:
sorry, I misunderstood your meaning. I thought you were referring to the existing modifications. If the original code poses risks, we will modify it. Otherwise, I prefer to keep this PR as minimal as possible rather than altering code unrelated to this PR.

Reply:
Oh, ok.

> If the original code poses risks, we will modify it.

It poses a risk although minimal. If `sizeof(unsigned long) > sizeof(void*)` and `sizeof(void*)` is 2 octets, then values greater than 65535 can't be carried by the `void*`, which I why I use a type that is as big as a `void*` can carry. The proper solution is of course to not use `void*` as a carrier of anyting but a pointer to the data needed by the thread, but that requires a bit more tinkering. My change is perhaps overkill since `BIO_WORKER_NUM` is currently `3` and will probably not grow much and fast. But it is nice to use a type that matches the carrier. Oh well... I'll remove it.

There's a runtime `serverAssert(worker < BIO_WORKER_NUM);` inside the thread I notice. Seems like overkill that too :-)

Reply:
@sundb A proper (i.m.h.) solution to get rid of the `void*` carrying the value of an `unsigned long` would be something like this: https://github.com/TedLyngmo/redis/commit/deefefd89552fafca992a93a36dc7ef6c3c19502

It also organizes things that belongs together nicely... i.m.h.😉